src/main/java/at/hannibal2/skyhanni/features/event/spook/TheGreatSpook.kt

```diff
@@ -44,20 +52,76 @@ object TheGreatSpook {
 
     @SubscribeEvent
     fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
-        if (isTimerEnabled()) config.positionTimer.renderString(displayTimer, posLabel = "Primal Fear Timer")
-        if (isFearStatEnabled()) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        if (config.primalFearTimer) config.positionTimer.renderString(displayTimer, posLabel = "Primal Fear Timer")
+        if (config.fearStatDisplay) {
             SkyblockStat.FEAR.displayValue?.let {
                 config.positionFear.renderString(it, posLabel = "Fear Stat Display")
             }
         }
-        if (isTimeLeftEnabled()) config.positionTimeLeft.renderString(displayTimeLeft, posLabel = "Time Left Display")
+        if (config.greatSpookTimeLeft) config.positionTimeLeft.renderString(displayTimeLeft, posLabel = "Time Left Display")
+    }
+
+    /**
+     * REGEX-TEST: §d§lQUICK MATHS! §r§7Solve: §r§e(10*2)+12*5
+     */
+    private val mathFearMessagePattern by RepoPattern.pattern(
+        "chat.math",
+        "§d§lQUICK MATHS! §r§7Solve: §r§e(?<math>.*)",
+    )
+
+    /**
+     * REGEX-TEST: §4[FEAR] Public Speaking Demon§r§f: Speak PlasticEating!
+     */
+    private val speakingFearMessagePattern by RepoPattern.pattern(
+        "chat.speaking",
+        "§4\\[FEAR] Public Speaking Demon§r§f: (Speak|Say something interesting) (?<name>.*)!",
+    )
+
+    private fun mathSolver(math: String?) {
+        val result = math?.let { NEUCalculator.calculateOrNull(it)?.toInt() } ?: run {
+            ChatUtils.userError("Failed to solve $math!")
+            return
+        }
+        ChatUtils.clickToActionOrDisable(
+            "The result is: $result",
+            config.primalFearSolver::math,
+            actionName = "Send the result",
+            action = {
+                HypixelCommands.allChat(result.toString())
+            },
+        )
     }
 
-    private fun isTimerEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearTimer
+    private fun publicSpeakingSolver() {
+        ChatUtils.clickToActionOrDisable(
+            "Click to send a random string to complete the Primal Fear",
```

Comment:
remove the phrase "to send a random string" from this line. otherwise the line gets too long also the whole chat message contains the information twice atm